test/docker-compose.e2e-tests.yml

```diff
@@ -5,46 +5,73 @@ version: "3.8"
+          graplctl \
+              --grapl-region $AWS_REGION \
+              --grapl-deployment-name $DEPLOYMENT_NAME \
+              --grapl-version $DEPLOYMENT_NAME \
+              --aws-profile default \
+              upload sysmon \
+              --logfile ./etc/sample_data/eventlog.xml && \
+          python -c 'import lambdex_handler; lambdex_handler.handler(None, None)'
       "
     volumes:
       - dynamodb_dump:/mnt/dynamodb_dump
       - ${PWD}/etc:/home/grapl/etc:ro
     environment:
-      - AWS_REGION
-      - DEPLOYMENT_NAME
-      - DEBUG_SERVICES=${DEBUG_SERVICES:-}
-      - DGRAPH_HOST
-      - DGRAPH_ALPHA_HTTP_EXTERNAL_PUBLIC_PORT
-      - DUMP_ARTIFACTS=${DUMP_ARTIFACTS:-False}
-      - DYNAMODB_ACCESS_KEY_ID
-      - DYNAMODB_ACCESS_KEY_SECRET
-      - DYNAMODB_ENDPOINT
-      - GRAPL_API_HOST
-      - GRAPL_HTTP_FRONTEND_PORT
-      - GRAPL_GRAPHQL_HOST
-      - GRAPL_GRAPHQL_PORT
-      - GRAPL_MODEL_PLUGIN_DEPLOYER_HOST
-      - GRAPL_MODEL_PLUGIN_DEPLOYER_PORT
-      - GRAPL_LOG_LEVEL
-      - IS_LOCAL=True
-      - MG_ALPHAS
-      - S3_ACCESS_KEY_ID
-      - S3_ACCESS_KEY_SECRET
-      - S3_ENDPOINT
-      - SQS_ACCESS_KEY_ID
-      - SQS_ACCESS_KEY_SECRET
-      - SQS_ENDPOINT
-      - VSC_DEBUGGER_PORT=${VSC_DEBUGGER_PORT_FOR_GRAPL_E2E_TESTS}
+      AWS_REGION: ${AWS_REGION}
+      DEPLOYMENT_NAME: ${DEPLOYMENT_NAME}
+      DEBUG_SERVICES: ${DEBUG_SERVICES:-}
+      GRAPL_LOG_LEVEL: ${GRAPL_LOG_LEVEL}
+      IS_LOCAL: "True"
+      S3_ACCESS_KEY_ID: ${S3_ACCESS_KEY_ID}
+      S3_ACCESS_KEY_SECRET: ${S3_ACCESS_KEY_SECRET}
+      S3_ENDPOINT: ${S3_ENDPOINT}
+      SQS_ACCESS_KEY_ID: ${SQS_ACCESS_KEY_ID}
+      SQS_ACCESS_KEY_SECRET: ${SQS_ACCESS_KEY_SECRET}
+      SQS_ENDPOINT: ${SQS_ENDPOINT}
+      SNS_ACCESS_KEY_ID: ${SNS_ACCESS_KEY_ID}
+      SNS_ACCESS_KEY_SECRET: ${SNS_ACCESS_KEY_SECRET}
+      SNS_ENDPOINT: ${SNS_ENDPOINT}
+      SSM_ACCESS_KEY_ID: ${SSM_ACCESS_KEY_ID}
+      SSM_ACCESS_KEY_SECRET: ${SSM_ACCESS_KEY_SECRET}
+      SSM_ENDPOINT: ${SSM_ENDPOINT}
+      SECRETSMANAGER_ACCESS_KEY_ID: ${SECRETSMANAGER_ACCESS_KEY_ID}
+      SECRETSMANAGER_ACCESS_KEY_SECRET: ${SECRETSMANAGER_ACCESS_KEY_SECRET}
+      SECRETSMANAGER_ENDPOINT: ${SECRETSMANAGER_ENDPOINT}
+      CLOUDWATCH_ACCESS_KEY_ID: ${CLOUDWATCH_ACCESS_KEY_ID}
+      CLOUDWATCH_ACCESS_KEY_SECRET: ${CLOUDWATCH_ACCESS_KEY_SECRET}
+      CLOUDWATCH_ENDPOINT: ${CLOUDWATCH_ENDPOINT}
+      LAMBDA_ACCESS_KEY_ID: ${LAMBDA_ACCESS_KEY_ID}
+      LAMBDA_ACCESS_KEY_SECRET: ${LAMBDA_ACCESS_KEY_SECRET}
+      LAMBDA_ENDPOINT: ${LAMBDA_ENDPOINT}
+      ROUTE53_ACCESS_KEY_ID: ${ROUTE53_ACCESS_KEY_ID}
+      ROUTE53_ACCESS_KEY_SECRET: ${ROUTE53_ACCESS_KEY_SECRET}
+      ROUTE53_ENDPOINT: ${ROUTE53_ENDPOINT}
+      VSC_DEBUGGER_PORT: ${VSC_DEBUGGER_PORT_FOR_GRAPL_E2E_TESTS}
+      MG_ALPHAS: ${MG_ALPHAS}
+      GRAPL_API_HOST: ${GRAPL_API_HOST}
+      GRAPL_HTTP_FRONTEND_PORT: ${GRAPL_HTTP_FRONTEND_PORT}
+      GRAPL_TEST_USER_NAME: ${GRAPL_TEST_USER_NAME}
```

Comment:
Environment variables defined with a map (as opposed to a list) can still inherit their values from the environment by leaving the value empty.

Why did you need to change them all from array to map, though?

https://docs.docker.com/compose/compose-file/compose-file-v3/#environment

Reply:
I believe at some point during the development of these changes I actually needed a mapping, but I can change it back now